[2023-12-26T11:07:29.384+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: final_project_dag.Stage_songs manual__2023-12-26T10:57:24.301119+00:00 [queued]>
[2023-12-26T11:07:29.395+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: final_project_dag.Stage_songs manual__2023-12-26T10:57:24.301119+00:00 [queued]>
[2023-12-26T11:07:29.395+0000] {taskinstance.py:2171} INFO - Starting attempt 3 of 4
[2023-12-26T11:07:29.415+0000] {taskinstance.py:2192} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2023-12-26 10:57:24.301119+00:00
[2023-12-26T11:07:29.424+0000] {standard_task_runner.py:60} INFO - Started process 16788 to run task
[2023-12-26T11:07:29.429+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'final_project_dag', 'Stage_songs', 'manual__2023-12-26T10:57:24.301119+00:00', '--job-id', '51', '--raw', '--subdir', 'DAGS_FOLDER/final_project/final_project.py', '--cfg-path', '/tmp/tmp56tdc8ub']
[2023-12-26T11:07:29.430+0000] {standard_task_runner.py:88} INFO - Job 51: Subtask Stage_songs
[2023-12-26T11:07:29.487+0000] {task_command.py:423} INFO - Running <TaskInstance: final_project_dag.Stage_songs manual__2023-12-26T10:57:24.301119+00:00 [running]> on host d3aa51350085
[2023-12-26T11:07:29.569+0000] {taskinstance.py:2481} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Sparkify' AIRFLOW_CTX_DAG_ID='final_project_dag' AIRFLOW_CTX_TASK_ID='Stage_songs' AIRFLOW_CTX_EXECUTION_DATE='2023-12-26T10:57:24.301119+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-12-26T10:57:24.301119+00:00'
[2023-12-26T11:07:29.570+0000] {stage_redshift.py:44} INFO - StageToRedshiftOperator not implemented yet
[2023-12-26T11:07:29.579+0000] {base.py:83} INFO - Using connection ID 'aws_credentials' for task execution.
[2023-12-26T11:07:29.580+0000] {connection_wrapper.py:378} INFO - AWS Connection (conn_id='aws_credentials', conn_type='aws') credentials retrieved from login and password.
[2023-12-26T11:07:29.597+0000] {stage_redshift.py:48} INFO - Clearing data from destination Redshift table
[2023-12-26T11:07:29.605+0000] {base.py:83} INFO - Using connection ID 'redshift' for task execution.
[2023-12-26T11:07:29.661+0000] {sql.py:432} INFO - Running statement: DELETE FROM staging_songs, parameters: None
[2023-12-26T11:07:29.666+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 49, in execute
    redshift.run("DELETE FROM {}".format(self.table))
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 392, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 437, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.UndefinedTable: relation "staging_songs" does not exist

[2023-12-26T11:07:29.676+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=final_project_dag, task_id=Stage_songs, execution_date=20231226T105724, start_date=20231226T110729, end_date=20231226T110729
[2023-12-26T11:07:29.689+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 51 for task Stage_songs (relation "staging_songs" does not exist
; 16788)
[2023-12-26T11:07:29.720+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2023-12-26T11:07:29.779+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
